.pylintrc

```diff
@@ -70,7 +70,8 @@ disable=missing-docstring,
       wrong-import-order, # Leave this up to isort
       bad-continuation, # Leave this up to black
       line-too-long, # Leave this up to black
-      exec-used
+      exec-used,
+      unsubscriptable-object
```

Comment:
I was getting `unsubscriptable-object` errors through the linter for the `self[0]` [lines](https://github.com/open-telemetry/opentelemetry-python/pull/1134/files#diff-36ec0e77c55d7f4a4caef71eda0b3191R187). I tried using `getattr` instead, but that gave me `mypy` errors for the attribute types. Couldn't find a better way to get around this change.

Reply:
Do you know what true negatives we end up with because of these changes? Ideally we don't reduce the robustness of the tooling significantly by making this small change. This relates to a comment I made on another PR I think in that it's important to make sure the SDK as a whole is optimized for, and not individual suggestions. Actually I've recently found the spec might have too much detail that makes it hard to reconcile with language-specific design like here - https://github.com/open-telemetry/opentelemetry-specification/pull/969. If immutable types aren't idiomatic in Python and cause our tooling to break, we need to consider whether it's actually worth it.

Reply:
Hmm, it's true that immutable types aren't idiomatic in Python. From @lzchen's comment below, it seems like it might be possible to disable `pylint` specifically for just the `self[0]` lines (as above) which I'll try first. However, it does seem like pylint does have a lot of false positive issues with the `unsubscriptable-object` error (like [here](https://github.com/PyCQA/pylint/issues/1498)).